torch/_subclasses/meta_utils.py

```diff
@@ -246,22 +246,174 @@ def sym_sizes_strides_storage_offset(
+                        )
+                        continue
+                else:
+                    # assume fully dynamic for other metadata (e.g. offsets, lengths)
+                    inner_policy = FreshCreateSymbolicPolicy(
+                        dynamic_sizes=[DimDynamic.DYNAMIC], constraint_sizes=[None]
+                    )
+
+                transformed_tensors_dict[attr] = callback(
+                    lambda: empty_create(
+                        inner_t,
+                        inner_src,
+                        inner_policy=inner_policy,
+                    )
+                )
+
+            # We expect JaggedTensor to have a 'ragged_size' in
+            # its context
+            assert isinstance(ctx, dict)
+            assert "ragged_size" in ctx
+            assert isinstance(t._size[1], torch.SymInt)
+            if orig_shape_env is shape_env:
+                # It's already fake and the shape envs line up, reuse the old size
+                # Do not assert singleton_int; it may already
+                # be a variable
+                ctx["ragged_size"] = t._size[1]
+            else:
+                assert (
+                    isinstance(t._size[1], torch.SymInt)
+                    and t._size[1].node.singleton_int() is not None
+                )
+                # Replace the eager ragged size with our freshly
+                # allocated jagged size that has a source
+
+                from torch._dynamo.source import TensorProperty, TensorPropertySource
+
+                # Replace the eager ragged size with our freshly
+                # allocated jagged size that has a source
+                ctx["ragged_size"] = shape_env.create_symintnode(
+                    shape_env.create_symbol(
+                        t._size[1],
+                        TensorPropertySource(
+                            source, TensorProperty.SIZE, 1  # type: ignore[arg-type]
+                        ),
+                    ),
+                    hint=t._size[1],
+                )
+
+            meta_nt = type(t).__tensor_unflatten__(transformed_tensors_dict, ctx)
+
+            if meta_base is None:
+                return meta_nt
+
+            # We have a dense base; maintain proper view relationship
+            from torch.nested._internal.nested_tensor import (
+                nested_view_from_values_offsets,
+                nested_view_from_values_offsets_lengths,
+            )
+
+            if meta_nt._lengths is None:
+                # Typical dense -> NT view case
+                return nested_view_from_values_offsets(
+                    meta_nt._values, meta_nt._offsets
+                )
+            else:
+                # base is a dense tensor that is the result of narrow()
+                return nested_view_from_values_offsets_lengths(
+                    meta_nt._values,
+                    meta_nt._offsets,
+                    meta_nt._lengths,
+                )
```

Comment:
Some of this code is the same as before, but it's really hard to tell because you are doing code motion and modification at the same time. To add insult to injury, this code is pretty hard to review from first principles because it's pretty convoluted